functional/vfio/run.sh

```diff
@@ -94,32 +98,63 @@ Usage: $0 [-h] [options]
     Description:
         This script runs a kata container and passthrough a vfio device
     Options:
-        -h,         Help
-        -s <value>, Set sandbox_cgroup_only in the configuration file
+        -h,          Help
+        -m <string>, Specify kata-runtime machine type for qemu hypervisor
+        -p <string>, Specify kata-runtime hypervisor
+        -s <value>,  Set sandbox_cgroup_only in the configuration file
 EOF
 }
 
 setup_configuration_file() {
+	clh_config_file="configuration-clh.toml"
+	qemu_config_file="configuration-qemu.toml"
+
 	for file in $(kata-runtime --kata-show-default-config-paths); do
-		if [ -f "${file}" ]; then
-			cp -a "${file}" "${CONFIG_FILE}"
+		config_dir=$(dirname ${file})
+		config_filename=$(basename ${file})
+
+		if [ "$HYPERVISOR" = "cloud-hypervisor" ]; then
+			config_filename="${clh_config_file}"
+		elif [ "$HYPERVISOR" = "qemu" ]; then
+			config_filename="${qemu_config_file}"
+		fi
+
+		config_file="${config_dir}/${config_filename}"
+
+		if [ -f "${config_file}" ]; then
+			cp -a "${config_file}" "${CONFIG_FILE}"
 			break
 		fi
 	done
 
+	# machine type applies to configuration.toml and configuration-qemu.toml
+	if [ "$HYPERVISOR" != "cloud-hypervisor" ]; then
+		if [ "$MACHINE_TYPE" = "pc" ] || [ "$MACHINE_TYPE" = "q35" ]; then
+			sed -i 's|^machine_type.*|machine_type = "'${MACHINE_TYPE}'"|g' "${CONFIG_FILE}"
+		fi
+	elif [ "$HYPERVISOR" = "cloud-hypervisor" ] && [ -n "$MACHINE_TYPE" ]; then
+		warn "Variable machine_type does not apply to cloud-hypervisor and will be ignored"
+	fi
+
```

Comment:
I think you can simplify this block

```sh
if [ -n "$MACHINE_TYPE" ]; then
    if [ "$HYPERVISOR" = "qemu" ]; then
        sed -i 's|^machine_type.*|machine_type = "'${MACHINE_TYPE}'"|g' "${CONFIG_FILE}"
    else
        warn "Variable machine_type only applies to qemu. It will be ignored"
    fi
fi
```

Reply:
@devimc updated. I originally had it checking that the hypervisor it wasn't cloud-hypervisor since the default config uses qemu, but it isn't obvious